test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/base/PipelineBaseE2EIT.java

```diff
@@ -383,4 +377,19 @@ protected void assertGreaterThanOrderTableInitRows(final int tableInitRows, fina
         int recordsCount = getTargetTableRecordsCount(tableName);
         assertTrue("actual count " + recordsCount, recordsCount > tableInitRows);
     }
+    
+    protected DataSource generateShardingSphereDataSourceFromProxy() throws SQLException {
```

Comment:
TODO could be added for `generateShardingSphereDataSourceFromProxy` method, about why it's needed.
And then it might be simplified later.